functions/esm/index.js

```diff
@@ -0,0 +1,23 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+// [START functions_esm]
```

Comment:
@bcoe I recall you saying the Node.js libraries weren't ready to support ESM yet, is that still the case? It seems okay if functions gets out ahead of the rest of Cloud, but we'll want to make the caveats highly visible if so.

In that case, I think this would need comments with that warning, in addition to carefully highlighting in the docs.

I assume if this is working the Functions Framework is ESM-ready?